pythran/pythonic/os/path/join.hpp

```diff
@@ -32,11 +38,15 @@ namespace pythonic
       void _join(types::str &buffer, T &&head, Types &&... tail)
       {
         if (head[0] == '/')
```

Comment:
Because for windows, we should handle the C:/ and not the / which is not supported here. I didn't want to spend too much time on it so I don't support all these possibilities. Because of this, the is_path_sep is not really interesting.